rpm/0084-sailfishos-gecko-Fix-download-starting-when-multiple.patch

```diff
@@ -0,0 +1,48 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Andrew den Exter <andrew.den.exter@jolla.com>
+Date: Fri, 5 Nov 2021 08:55:22 +0000
+Subject: [PATCH] [sailfishos][gecko] Fix download starting when multiple tabs
+ are closed. JB#56047 OMP#JOLLA-476
+
+When the active tab is closed another tab will be made active and start
+to load, if that tab is closed while loading the parent puppet widget can be
+destroyed before the content view is created causing a failure in the
+content view construction the error from which gets interpreted as the
+unsupported document type and the browser falls back to downloading.
+
+There is a failure case for when this happens and the DocShell is in
+destroyed state which cancels the loading the document, so we can do the
+same if the puppet widget has been destroyed.
+---
+ docshell/base/nsDocShell.cpp | 8 ++++++--
+ 1 file changed, 6 insertions(+), 2 deletions(-)
+
+diff --git a/docshell/base/nsDocShell.cpp b/docshell/base/nsDocShell.cpp
+index ae8b77a80cdc..912ca0b67a5a 100644
+--- a/docshell/base/nsDocShell.cpp
++++ b/docshell/base/nsDocShell.cpp
+@@ -7659,7 +7659,8 @@ nsresult nsDocShell::CreateContentViewer(const nsACString& aContentType,
+     }
+   }
+ 
+-  NS_ENSURE_SUCCESS(Embed(viewer), NS_ERROR_FAILURE);
++  rv = Embed(viewer);
++  NS_ENSURE_SUCCESS(rv, rv);
+ 
+   if (!mBrowsingContext->GetHasLoadedNonInitialDocument()) {
+     mBrowsingContext->SetHasLoadedNonInitialDocument(true);
+@@ -7853,7 +7854,10 @@ nsresult nsDocShell::SetupNewViewer(nsIContentViewer* aNewViewer,
+ 
+   mContentViewer->SetNavigationTiming(mTiming);
+ 
+-  if (NS_FAILED(mContentViewer->Init(widget, bounds, aWindowActor))) {
++  if (widget->Destroyed()) {
```

Comment:
```suggestion
+  if (widget && widget->Destroyed()) {
```
That's bare minimum needed change. I'd try to check if could destroy and clear docshell and friends in the EmbedLiteViewChild::RecvDestroy

Reply:
Calling mWebBrowser->Destroy(); in RecvDestroy worked for me closing a dozen tabs in a row. And I put some debug in nsDocShell::Destroy() and saw that it was being called. I also tried just clearing the mWebBrowser pointer but I didn't see my debug statements and then I had a crash.

Reply:
Calling mWebBrowser->Destroy(); in RecvDestroy of EmbedLiteViewChild a assume. If so, that would be better as then we can avoid one additional patch.